### PR TITLE
Updates.

### PR DESCRIPTION
**Pull request by groverHarshit**
All the updates will be listed under this PR

### CODE REVIEW

File: api/controllers/standard/standard.controller.js [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+            }else{
+                {
+                    const api_auth_key = data.api_auth_key;
+                    delete data['api_auth_key'];
+                    if(process.env.api_auth_key === api_auth_key){
+                        await Standard.create(data)
+                            .then(result => {
+                                res.json({
+                                    status:200,
+                                    message: 'Insert Sucessfull'
```
Comment (upperwal) [inline api/controllers/standard/standard.controller.js:19]: 'Successful'

File: api/controllers/standard/standard.controller.js [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+                                });
+                            })
+                            .catch(err => {
+                                res.json({
+                                    status:404,
+                                    message:"Insert Unsucesessfull",
```
Comment (upperwal) [inline api/controllers/standard/standard.controller.js:25]: A Better message could be: 'Standard creation failed'

File: api/controllers/standard/standard.controller.js (hunk 1 above)
Comment (groverHarshit) [reply api/controllers/standard/standard.controller.js:19]: Noted

File: api/controllers/standard/standard.controller.js (hunk 2 above)
Comment (groverHarshit) [reply api/controllers/standard/standard.controller.js:25]: Noted

File: api/controllers/standard/standard.controller.js [hunk 3]
```diff
@@ -1,31 +1,160 @@
...
+  const increment_standard = req.body.increment_standard;
+  await Standard.find({ "info.shortName": shortName })
+    .then(async (result) => {
+      if (result.length > 0) {
+        result = result[0];
+        result["versions"] = [
+          result["versions"][result["versions"].length - 1],
+        ];
+        version = result["versions"][0].version;
+        version[`${increment_standard}`]++;
```
Comment (upperwal) [inline api/controllers/standard/standard.controller.js:124]: We will be following the (semantic versioning standard](https://semver.org) which has 3 parts `MAJOR.MINOR.PATH`. I think the user should pass the kind of update they are doing to the standard (`MAJOR, MINOR or PATCH`) and based on that we should increment the fields.

So, now your `version` object in `standard.model` would look something like this (which is done in this patch):

```json
version: {
  "major": 1,
  "minor": 2,
  "patch": 3
}
```

Based on the `kindOfUpdate` you can increment the fields.

```py
if kindOfUpdate == 'MAJOR':
  new_version = {
    "major": old_version.major + 1,
    "minor": 0,
    "patch": 0
  }
elif kindOfUpdate == 'MINOR':
  new_version = {
    "major": old_version.major,
    "minor": old_version.minor + 1,
    "patch": 0
  }
elif kindOfUpdate == 'PATCH':
  new_version = {
    "major": old_version.major,
    "minor": old_version.minor,
    "patch": old_version.patch + 1
  }
```

File: api/controllers/standard/standard.controller.js [hunk 4]
```diff
@@ -1,31 +1,160 @@
...
+        versions["version"] = version;
+        await Standard.update({ $push: { versions: versions } })
```
Comment (upperwal) [inline api/controllers/standard/standard.controller.js:126]: Also need to change the update timestamp

**Review by upperwal — CHANGES_REQUESTED** [review]

**Comment by groverHarshit** [comment]
@upperwal changes are pushed and ready.
please verify at your end

**Comment by upperwal** [comment]
Thanks @groverHarshit

**Pull request merged**